language/diem-vm/src/diem_vm.rs

```diff
@@ -468,54 +469,58 @@ impl<'a> DiemVMInternals<'a> {
     }
 }
 
-pub fn txn_effects_to_writeset_and_events_cached<C: AccessPathCache>(
+pub fn convert_changeset_and_events_cached<C: AccessPathCache>(
     ap_cache: &mut C,
-    effects: TransactionEffects,
+    changeset: MoveChangeSet,
+    events: Vec<MoveEvent>,
 ) -> Result<(WriteSet, Vec<ContractEvent>), VMStatus> {
     // TODO: Cache access path computations if necessary.
     let mut ops = vec![];
 
-    for (addr, vals) in effects.resources {
-        for (struct_tag, val_opt) in vals {
+    for (addr, account_changeset) in changeset.accounts {
+        for (struct_tag, blob_opt) in account_changeset.resources {
             let ap = ap_cache.get_resource_path(addr, struct_tag);
-            let op = match val_opt {
+            let op = match blob_opt {
                 None => WriteOp::Deletion,
-                Some((ty_layout, val)) => {
-                    let blob = val.simple_serialize(&ty_layout).ok_or(VMStatus::Error(
-                        StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
-                    ))?;
-
-                    WriteOp::Value(blob)
-                }
+                Some(blob) => WriteOp::Value(blob),
             };
             ops.push((ap, op))
         }
-    }
 
-    for (module_id, blob) in effects.modules {
-        ops.push((ap_cache.get_module_path(module_id), WriteOp::Value(blob)))
+        for (name, blob_opt) in account_changeset.modules {
+            let ap = ap_cache.get_module_path(ModuleId::new(addr, name));
```

Comment:
Now that we don't hash access paths anymore, we don't need `ap_cache` at all--this can just be `AccessPath::from(ModuleId::new(addr, name))`.

We should probably refactor to get rid of `AccessPathCache` altogether to get rid of all the indirection it adds CC @runtian-zhou

Reply:
Will put up a follow-up PR for this.